[Bug] `InaccessibleObjectException` with generic enum fields in EqualsVerifier 4.x

### Describe the bug

I've encountered an issue where EqualsVerifier 4.1 throws `java.lang.reflect.InaccessibleObjectException` when verifying classes with generic enum fields (`E extends Enum<E>`). This affects both direct generic enum containers and classes that inherit generic enum fields from superclasses.
This works correctly in EqualsVerifier 3.x but seems to fail in all 4.x versions I've tested.

### Steps to reproduce

1. Create a class with a generic enum field `E extends Enum<E>`, or
2. Create a class that extends a superclass with a generic enum field
3. Run EqualsVerifier on the class
4. The test fails with `InaccessibleObjectException`

### Error message and version number


**EqualsVerifier Version**: 4.1
**Java Version**: 21.0.5

```
EqualsVerifier found a problem in class com.example.Dummy.
-> Field state of type java.lang.Enum is not accessible via the Java Module System.
Consider opening the module that contains it, or add prefab values for type java.lang.Enum.

java.lang.AssertionError
Caused by: nl.jqno.equalsverifier.internal.exceptions.ModuleException
Caused by: nl.jqno.equalsverifier.internal.exceptions.ModuleException
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make field private final java.lang.String java.lang.Enum.name accessible: module java.base does not "opens java.lang" to unnamed module
```

### Code: EqualsVerifier invocation

```java
import nl.jqno.equalsverifier.EqualsVerifier;
import org.junit.jupiter.api.Test;

class DummyTest {
    @Test
    void has_correct_equals_and_hashcode() {
        EqualsVerifier.forClass(Dummy.class)
            .usingGetClass()
            .verify();
    }
}
```

### Code: class under test

```java
import java.util.Objects;

// Base class with generic enum field
class StatefulEntity<S extends Enum<S>> {
    
    public final S state;

    protected StatefulEntity(S state) {
        this.state = state;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        StatefulEntity<?, ?> statefulEntity = (StatefulEntity<?, ?>) o;
        return Objects.equals(state, statefulEntity.state);
    }

    @Override
    public int hashCode() {
        return Objects.hash(state);
    }
}

// Concrete implementation
class Dummy extends StatefulEntity<Dummy.State> {
    private final String value;

    public Dummy(State state, String value) {
        super(state);
        this.value = value;
    }

    @Override
    public boolean equals(Object o) {
        if (o == null || getClass() != o.getClass()) return false;
        if (!super.equals(o)) return false;
        Dummy someModel = (Dummy) o;
        return Objects.equals(value, someModel.value);
    }

    @Override
    public int hashCode() {
        return Objects.hash(super.hashCode(), value);
    }

    public enum State {
        A, B
    }
}
```


### Additional context

This appears to be a regression introduced in EqualsVerifier 4.0, as the same code works correctly with version 3.19.4.

Thank you for your time and consideration in looking into this issue!

--- a/equalsverifier-core/src/main/java/nl/jqno/equalsverifier/internal/instantiation/prefab/BuiltinPrefabValueProvider.java
+++ b/equalsverifier-core/src/main/java/nl/jqno/equalsverifier/internal/instantiation/prefab/BuiltinPrefabValueProvider.java
@@ -18,7 +18,7 @@ import nl.jqno.equalsverifier.internal.util.PrimitiveMappers;
 public class BuiltinPrefabValueProvider implements ValueProvider {
 
     private static final Set<Class<?>> EXCEPTIONAL_GENERIC_TYPES =
-            Set.of(Class.class, Constructor.class, SynchronousQueue.class);
+            Set.of(Class.class, Constructor.class, SynchronousQueue.class, Enum.class);
 
     /** {@inheritDoc}} */
     @Override

--- a/equalsverifier-test/src/test/java/nl/jqno/equalsverifier/integration/extended_contract/GenericTypesTest.java
+++ b/equalsverifier-test/src/test/java/nl/jqno/equalsverifier/integration/extended_contract/GenericTypesTest.java
@@ -7,6 +7,7 @@ import java.util.function.Supplier;
 import nl.jqno.equalsverifier.EqualsVerifier;
 import nl.jqno.equalsverifier.Warning;
 import nl.jqno.equalsverifier_testhelpers.ExpectedException;
+import nl.jqno.equalsverifier_testhelpers.types.Color;
 import nl.jqno.equalsverifier_testhelpers.types.Point;
 import nl.jqno.equalsverifier_testhelpers.types.SparseArrays.SparseArrayEqualsContainer;
 import nl.jqno.equalsverifier_testhelpers.types.SparseArrays.SparseArrayHashCodeContainer;
@@ -122,6 +123,16 @@ class GenericTypesTest {
     @Test
     void succeed_whenFieldHasARecursiveGenericAndAWildcard() {
         EqualsVerifier.forClass(RecursiveGenericWithWildcardContainer.class).verify();
+    }
+
+    @Test
+    void succeed_whenClassHasGenericEnumField() {
+        EqualsVerifier.forClass(GenericEnumContainer.class).usingGetClass().verify();
+    }
+
+    @Test
+    void succeed_whenClassInheritsGenericEnumField() {
+        EqualsVerifier.forClass(ConcreteColorContainer.class).usingGetClass().verify();
     }
 
     static final class GenericContainerWithBuiltin {
@@ -624,6 +635,56 @@ class GenericTypesTest {
         @Override
         public int hashCode() {
             return Objects.hash(rg);
+        }
+    }
+
+    static class GenericEnumContainer<E extends Enum<E>> {
+
+        protected final E ev;
+
+        protected GenericEnumContainer(E ev) {
+            this.ev = ev;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            GenericEnumContainer<?> that = (GenericEnumContainer<?>) o;
+            return Objects.equals(ev, that.ev);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hashCode(ev);
+        }
+    }
+
+    static final class ConcreteColorContainer extends GenericEnumContainer<Color> {
+
+        private final int i;
+
+        public ConcreteColorContainer(Color color, int i) {
+            super(color);
+            this.i = i;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            if (!super.equals(o)) {
+                return false;
+            }
+            ConcreteColorContainer that = (ConcreteColorContainer) o;
+            return i == that.i;
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(super.hashCode(), i);
         }
     }
 }

